echopype/clean/impulse_noise.py

```diff
@@ -0,0 +1,463 @@
+    the wang from echopy function. Users should take into
+    account that regions marked as True in the returned mask have been identified as noise.
+
+
+    """
+
+    # Select the desired frequency channel directly using 'sel'
+    selected_channel_ds = Sv_ds.sel(channel=desired_channel)
+
+    # Extract Sv values for the desired frequency channel
+    Sv = selected_channel_ds["Sv"].values
+
+    # Check if there are any NaN values in the Sv data
+    if np.isnan(Sv).any():
+        warnings.warn(
+            "Input Sv data contains NaN values."
+            "These values are not processed correctly by the impulse noise removal algorithm"
+            "and will be marked as noise in the output mask."
+            "Please ensure that Sv data is cleaned or appropriately "
+            "preprocessed before using this function."
+        )
+
+    # Transpose the Sv data so that the vertical dimension is the first dimension (axis 0)
+    Sv = np.transpose(Sv)
+
+    """
+    Call the wang function to get the cleaned Sv data and the mask indicating edges,
+    where swarms analysis couldn't be performed
+    The variable mask_ is a boolean mask where
+    True represents edges where cleaning wasn't applied,
+    and False represents areas where cleaning was applied
+    """
+    Sv_cleaned, mask_ = _wang(Sv, thr, erode, dilate, median)
+
+    """
+    Create a boolean mask comparing the original and cleaned Sv data
+    Creates a boolean mask where True denotes locations where the original Sv values
+    are different from the cleaned Sv values.
+    """
+
+    noise_mask = Sv != Sv_cleaned
+
+    # Combined mask
+    # The bitwise negation ~ operator is applied to mask_.
+    # So, ~mask_ is True where cleaning was applied and
+    # False where cleaning wasn't applied (the edges).
+    combined_mask = np.logical_and(~mask_, noise_mask)
+
+    # Transpose the mask back to its original shape
+    # Combined_mask is a mask that marks valid (non-edge) locations where
+    # noise has been identified and cleaned.
+    combined_mask = np.transpose(noise_mask)
+
+    # Create a new xarray for the mask with the correct dimensions and coordinates
+    mask_xr = xr.DataArray(
+        combined_mask,
+        dims=("ping_time", "range_sample"),
+        coords={
+            "ping_time": selected_channel_ds.ping_time.values,
+            "range_sample": selected_channel_ds.range_sample.values,
+        },
+    )
+
+    warnings.warn(
+        "The output mask from this function identifies regions of noise in the Sv data, "
+        "but does not modify them in the same way as the `wang` function from echopy."
+        "Visualizations using this mask may therefore differ from"
+        "those generated using the `wang` function from echopy. "
+        "Be aware that regions marked as True in the mask are identified as noise."
+    )
+
+    return mask_xr
+
+
+def _find_impulse_mask_ryan(
```

Comment:
Once you modify `_ryan` under the hood to use xarray operations, I think you could remove this function and just make sure output of `_ryan` is of the right dimensions.

Reply:
https://github.com/OceanStreamIO/echopype/commit/4294926d3732bc1a0bb574331bc6afcb6ffea7d4 consolidated into a single function